native/cocos/ui/edit-box/EditBox-ios.mm

```diff
@@ -363,76 +267,331 @@ - (BOOL)textFieldShouldReturn:(UITextField *)textField {
+    textView.backgroundColor = [UIColor whiteColor];
+    textView.layer.cornerRadius = 5.0;
+    textView.clipsToBounds = YES;
+    textView.text = [NSString stringWithUTF8String:showInfo->defaultValue.c_str()];
+    TextViewDelegate* delegate = [[TextViewDelegate alloc] initWithPairs:[inputbox inputOnView] and:textView];
+    inputbox.inputDelegate = delegate;
+    textView.delegate = delegate;
+    UIBarButtonItem *textViewItem = [[UIBarButtonItem alloc] initWithCustomView:textView];
+    
+    if (!btnHandler){
+        btnHandler = [[ButtonHandler alloc] init];
+    }
+    UIButton *confirmBtn = [[UIButton alloc]
+                            initWithFrame:CGRectMake(0,
+                                                     0,
+                                                     BUTTON_WIDTH,
+                                                     BUTTON_HEIGHT)];
+    [confirmBtn addTarget:btnHandler
+                   action:@selector(buttonTapped:)
+         forControlEvents:UIControlEventTouchUpInside];
+    [confirmBtn setTitle:[NSString stringWithUTF8String:showInfo->confirmType.c_str()]
+             forState:UIControlStateNormal];
+    [confirmBtn setTitleColor:[UIColor systemBlueColor]
+                  forState:UIControlStateNormal];
+    [confirmBtn setTitleColor:[UIColor darkTextColor]
+                  forState:UIControlStateHighlighted]; // Hight light state triggered when the button is tapped.
+    UIBarButtonItem *confirm = [[UIBarButtonItem alloc]initWithCustomView:confirmBtn];
+    
+    [toolbar setItems:@[textViewItem, confirm] animated:YES];
+    UIBarButtonItem* textViewBarButtonItem = [self setInputWidthOf:toolbar];
+    ((UITextView*)[inputbox inputOnView]).inputAccessoryView = toolbar;
+    
+    [inputbox setInputOnToolbar:textViewBarButtonItem.customView];
+    //release for NON ARC ENV
+    [toolbar release];
+    [textView release];
+    [confirmBtn release];
+    [textViewItem release];
+    [confirm release];
+    
+}
+- (void) addInputAccessoryViewForTextField: (InputBoxPair*)inputbox
+                                      with: (const cc::EditBox::ShowInfo*)showInfo{
+    CGRect safeView = getSafeAreaRect();
+    UIToolbar* toolbar = [[UIToolbar alloc]
+                          initWithFrame:CGRectMake(0,
+                                                   0,
+                                                   safeView.size.width,
+                                                   TEXT_LINE_HEIGHT + ITEM_MARGIN_HEIGHT)];
+    [toolbar setBackgroundColor:[UIColor darkGrayColor]];
+    
+    UITextField* textField = [[UITextField alloc] init];
+    textField.borderStyle = UITextBorderStyleRoundedRect;
+    textField.textColor = [UIColor blackColor];
+    textField.backgroundColor = [UIColor whiteColor];
+    textField.text = [NSString stringWithUTF8String:showInfo->defaultValue.c_str()];
+    TextFieldDelegate* delegate = [[TextFieldDelegate alloc] initWithPairs:[inputbox inputOnView] and:textField];
```

Comment:
where is `delegate` released ?

Reply:
> As it might be recreated again and again

So to prevent this behavior, just keep the week reference in InputBox.

But yes, the delegate should be release to decrease the retainCount once Inputbox is deleted